src/ui/components/SecondaryToolbox/ReactDevTools.tsx

```diff
@@ -34,28 +41,70 @@ class ReplayWall implements Wall {
 
   // called by the frontend to send a request to the backend
   async send(event: string, payload: any) {
-    if (event == "inspectElement") {
-      if (this.inspectedElements.has(payload.id) && !payload.path) {
-        // this element has been inspected before, the frontend asks to inspect it again
-        // to see if there are any changes - in Replay there won't be any so we can send
-        // the response immediately without asking the backend
-        this._listener?.({
-          event: "inspectedElement",
-          payload: {
-            responseID: payload.requestID,
-            id: payload.id,
-            type: "no-change",
-          },
-        });
-      } else {
-        if (!payload.path) {
-          this.inspectedElements.add(payload.id);
+    switch (event) {
+      case "inspectElement": {
+        if (this.inspectedElements.has(payload.id) && !payload.path) {
+          // this element has been inspected before, the frontend asks to inspect it again
+          // to see if there are any changes - in Replay there won't be any so we can send
+          // the response immediately without asking the backend
+          this._listener?.({
+            event: "inspectedElement",
+            payload: {
+              responseID: payload.requestID,
+              id: payload.id,
+              type: "no-change",
+            },
+          });
+        } else {
+          if (!payload.path) {
+            this.inspectedElements.add(payload.id);
+          }
+          this.sendRequest(event, payload);
         }
+        break;
+      }
+
+      case "getBridgeProtocol": {
         this.sendRequest(event, payload);
+        break;
+      }
+
+      case "highlightNativeElement": {
+        const { rendererID, id } = payload;
+
+        if (this.highlightedElementId) {
+          Highlighter.unhighlight();
+        }
+        this.highlightedElementId = { rendererID, id };
+
+        const response = await ThreadFront.evaluate(
+          0,
+          undefined,
+          `__REACT_DEVTOOLS_GLOBAL_HOOK__.rendererInterfaces.get(${rendererID}).findNativeNodesForFiberID(${id})[0]`
+        );
+        if (!response.returned || !this.isHighlightedElementID(rendererID, id)) {
+          return;
+        }
+
+        const nodeFront = response.returned.getNodeFront();
+        if (!nodeFront) {
+          return;
+        }
+
+        await nodeFront.ensureLoaded();
+        if (!this.isHighlightedElementID(rendererID, id)) {
+          return;
+        }
+
```

Comment:
Why three separate early returns along with the duplicate `isHighlightedElementId()` call? Maybe simplify to

```ts
        const nodeFront = response.returned?.getNodeFront();
        await nodeFront?.ensureLoaded();
        if (!nodeFront || !this.isHighlightedElementID(rendererID, id)) {
          return;
        }
```

Reply:
`isHighlightedElement()` is used to check if the user still hovers over the same component in ReactDevTools. If he doesn't, we want to avoid unnecessary work (`nodeFront.ensureLoaded()` sends 5 requests to the backend). Imagine if the user quickly moves the mouse across the component tree, hitting 20 components: without the first `isHighlightedElement()` check, we'd be sending 100 unnecessary requests to the backend, with the first check most of that work will be avoided.
The `if (!nodeFront)` check isn't necessary though, so I simplified it as suggested.